src/gp/kernel.jl

```diff
@@ -106,15 +108,20 @@ The usual periodic kernel derived by mapping the input domain onto the unit circ
 
 For length scales etc see [`stretch`](@ref), for variance see [`*`](@ref).
 """
-struct PerEQ <: Kernel end
+struct PerEQ{LT<:AV{<:Real}} <: Kernel
+	logl::LT
+end
+@functor PerEQ
+
+_pereq(d, logl) = exp(-2.0*sin(π*d)^2 / exp(2.0*logl))
 
 # Binary methods.
-ew(k::PerEQ, x::AV{<:Real}, x′::AV{<:Real}) = exp.(.-2 .* sin.(π .* abs.(x .- x′)).^2)
-pw(k::PerEQ, x::AV{<:Real}, x′::AV{<:Real}) = exp.(.-2 .* sin.(π .* abs.(x .- x′')).^2)
+ew(k::PerEQ, x::AV, x′::AV) = _pereq.(ew(Euclidean(), x, x′), k.logl[1]) 
+pw(k::PerEQ, x::AV, x′::AV) = _pereq.(pw(Euclidean(), x, x′), k.logl[1])
 
 # Unary methods.
-ew(::PerEQ, x::AV{<:Real}) = ones(eltype(x), length(x))
-pw(k::PerEQ, x::AV{<:Real}) = pw(k, x, x)
+ew(k::PerEQ, x::AV) = _pereq.(ew(Euclidean(), x), k.logl[1])
```

Comment:
Why has this been changed?

Reply:
The `PerEQ` should have three parameters: `σ²` ( scaling factor ), `l` ( length scale ) & `p` ( period ), using `Scaled` & `Stretched` can specify the first two, but I don't find way for the third one ( maybe Stheno already has it implemented, please indicate to me if I miss it ). 

As is mentioned [here](https://www.cs.toronto.edu/~duvenaud/cookbook/), the formula for periodic kernel looks like ( without scaling & length factor ):
```julia
exp(-2.0 * sin(π * d)^2 / l^2)
```

While Stheno's current implementation is:
```julia
exp.(.-2 .* sin.(π .* abs.(x .- x′)).^2)
```
and one can only pass in `AbstractVector` type to  `ew` & `pw` method for `PerEQ`, when pass in `ColVecs`, I got the following error message:
```julia
julia> pereq = Stheno.PerEQ()
julia> x = rand(3, 10);
julia> pw(pereq, ColVecs(x), ColVecs(x))
ERROR: MethodError: no method matching pairwise(::Stheno.PerEQ, ::ColVecs{Float64,Array{Float64,2}}, ::ColVecs{Float64,Array{Float64,2}})
```
also, Stheno doesn't explicitly export the `PerEQ`.

> I'm using Stheno v0.6.1 with julia v1.3.1 under macOS